engine/class_modules/paladin/sc_paladin.cpp

```diff
@@ -3284,39 +3284,67 @@ std::unique_ptr<expr_t> paladin_t::create_expression( util::string_view name_str
     cooldown_t* j_cd;
     cooldown_t* how_cd;
     cooldown_t* wake_cd;
+    cooldown_t* hs_cd;
+    cooldown_t* at_cd;
 
     time_to_hpg_expr_t( util::string_view n, paladin_t& p )
       : paladin_expr_t( n, p ),
         cs_cd( p.get_cooldown( "crusader_strike" ) ),
         boj_cd( p.get_cooldown( "blade_of_justice" ) ),
         j_cd( p.get_cooldown( "judgment" ) ),
         how_cd( p.get_cooldown( "hammer_of_wrath" ) ),
-        wake_cd( p.get_cooldown( "wake_of_ashes" ) )
+        wake_cd( p.get_cooldown( "wake_of_ashes" ) ),
+        hs_cd( p.get_cooldown( "holy_shock" ) ),
+        at_cd( p.get_cooldown( "arcane_torrent" ) )
     {
     }
 
     double evaluate() override
     {
-      if ( paladin.specialization() != PALADIN_RETRIBUTION )
+      if ( paladin.specialization() == PALADIN_PROTECTION )
       {
-        paladin.sim->errorf( "\"time_to_hpg\" only supported for Retribution" );
+        paladin.sim->errorf( "\"time_to_hpg\" not supported for Protection" );
```

Comment:
I think this whole check should be either moved to the expression constructor (throw an exception), or removed completely (by providing a fallback value for non protection paladins).

Having a sim error in a expression during runtime is strange, and it will only trigger once that code is reached, which might be hours into a profileset sim or similar.

Reply:
i think thats fair, i just did this to edit the code in place rather than fully refactor how it should be :) would prefer to fix the bug first and then update with a followup commit with the format/style change

Reply:
Yeah, agree, let's do that as a followup